app/services/stage_files.rb

```diff
@@ -6,17 +6,34 @@ def self.stage(signed_ids, druid)
     # Skip side effects if no signed IDs provided
     return yield if signed_ids.empty?
 
+    @dir = StagingDirectory.new(druid: druid, staging_location: Settings.staging_location)
```

Comment:
It could be a follow-up to this PR, but given that 1) the number of methods in this class has grown, and 2) we are holding onto state (`@dir`), I wonder if we should consider refactoring this class so that `StageFiles.stage` remains the entry but the rest of the work is done at the instance level?